datacats/task.py

```diff
@@ -0,0 +1,219 @@
+# Copyright 2014-2015 Boxkite Inc.
+
+# This file is part of the DataCats package and is released under
+# the terms of the GNU Affero General Public License version 3.0.
+# See LICENSE.txt or http://www.fsf.org/licensing/licenses/agpl-3.0.html
+
+"""
+This modle includes the implementations for many Environment methods
+separated from their configuration object to make them easier to test.
+"""
+
+# import modules not names inside so tests can mock functions
+# inside these modules more easily
+import os
+from os import path
+import ConfigParser
+
+from datacats import docker, validate
+from datacats.error import DatacatsError
+
+
+def list_sites(datadir):
+    """
+    Return a list of the site names valid for this environment
+    """
+    try:
+        return os.listdir(datadir + '/sites')
+    except OSError:
+        return []
+
+
+def save_site(site_name, sitedir, srcdir, port, address, site_url, passwords):
+    """
+    Add a site's configuration to the source dir and site dir
+    """
+    cp = ConfigParser.SafeConfigParser()
+    cp.read([srcdir + '/.datacats-environment'])
+
+    section_name = 'site_' + site_name
+
+    cp.add_section(section_name)
+    cp.set(section_name, 'port', str(port))
+    cp.set(section_name, 'address', address or '127.0.0.1')
+
+    if site_url:
+        cp.set(section_name, 'site_url', site_url)
+
+    with open(srcdir + '/.datacats-environment', 'w') as config:
+        cp.write(config)
+
+    # save passwords to datadir
+    cp = ConfigParser.SafeConfigParser()
+
+    cp.add_section('passwords')
+    for n in sorted(passwords):
+        cp.set('passwords', n.lower(), passwords[n])
+
+    # Write to the sitedir so we maintain separate passwords.
+    with open(sitedir + '/passwords.ini', 'w') as config:
+        cp.write(config)
+
+
+def save(name, datadir, srcdir, ckan_version, deploy_target=None,
```

Comment:
should `save` be renamed to `save_environment` here to match `save_site` naming?